### PR TITLE
[Refactor]: Refactor CAE

### PR DESCRIPTION
**Pull request by YuanLiuuuuuu**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: configs/selfsup/_base_/datasets/imagenet_cae.py [hunk 1]
```diff
@@ -1,7 +1,15 @@
 # dataset settings
 dataset_type = 'mmcls.ImageNet'
 data_root = 'data/imagenet/'
-file_client_args = dict(backend='disk')
+# file_client_args = dict(backend='disk')
```
Comment (fangyixiao18) [inline configs/selfsup/_base_/datasets/imagenet_cae.py]: change back to disk

File: tests/test_models/test_algorithms/test_cae.py [hunk 2]
```diff
@@ -23,10 +23,15 @@
...
-    tokenizer_path='cae_ckpt/encoder_stat_dict.pth',
-    loss=dict(type='CAELoss', lambd=2))
+head = dict(type='CAEHead', loss=dict(type='CAELoss', lambd=2))
+
+target_generator = dict(
+    type='DALL-E',
+    init_cfg=dict(
+        type='Pretrained',
+        checkpoint=  # noqa: E251
+        'https://download.openmmlab.com/mmselfsup/1.x/target_generator_ckpt/dalle_encoder.pth',  # noqa: E501
```
Comment (fangyixiao18) [inline tests/test_models/test_algorithms/test_cae.py]: it is no need to download ckpt in ut

**Comment by fangyixiao18** [comment]
related issue: https://github.com/open-mmlab/mmselfsup/issues/571

**Comment by YuanLiuuuuuu** [comment]
> related issue: #571
The refactored model is being trained. And I will create a new PR to update the link, once avaliable.

**Review by fangyixiao18 — APPROVED** [review]

**Review by tonysy — APPROVED** [review]
LGTM

**Pull request merged**